### PR TITLE
Only support containers stats using cgroups v2

### PR DESCRIPTION
**Pull request by jwhonce**
[NO TESTS NEEDED]

Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1988252

Signed-off-by: Jhon Honce <jhonce@redhat.com>

### CODE REVIEW

File: pkg/api/handlers/libpod/containers_stats.go [hunk 1]
```diff
@@ -20,6 +21,12 @@ func StatsContainer(w http.ResponseWriter, r *http.Request) {
 	runtime := r.Context().Value("runtime").(*libpod.Runtime)
 	decoder := r.Context().Value("decoder").(*schema.Decoder)
 
+	if isV2, _ := cgroups.IsCgroup2UnifiedMode(); !isV2 {
```
Comment (Luap99) [inline pkg/api/handlers/libpod/containers_stats.go:24]: This is only the case for rootless. Rootful should work with cgroup v2

**Review by vrothberg — COMMENTED** [review]
I wonder why nothing had to be changed in CI. Are we not exercising that?

**Comment by mheon** [comment]
@vrothberg I bet we deliberately disabled the tests for v1, because this never worked. We could add a negative test, potentially?

**Comment by vrothberg** [comment]
> @vrothberg I bet we deliberately disabled the tests for v1, because this never worked. We could add a negative test, potentially?

Yes, that sounds like a good idea.

**Comment by jwhonce** [comment]
> I wonder why nothing had to be changed in CI. Are we not exercising that?

@vrothberg I added an API test, that now covers expected return codes for root cgroup V1 and V2 vs rootless cgroup V1 or V2. Assuming that the CI runs through all those iterations. /cc @edsantiago @cevich

**Review by vrothberg — COMMENTED** [review]
/lgtm
/hold
@containers/podman-maintainers PTAL

**Pull request merged**